plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -524,6 +524,11 @@ func (c *X509Cert) collectCertURLs() []*url.URL {
 	return urls
 }
 
+func getSerialNumberString(cert *x509.Certificate) string {
+	// Rather than calling '.Text(16)' this retains leading zeroes
+	return hex.EncodeToString(cert.SerialNumber.Bytes())
+}
+
```

Comment:
I think we cannot do this unconditionally without breaking existing users as we change the tag value for current users with serial numbers containing leading zeros.

Furthermore, I would then suggest to add the leading zeros up to at beast 128-bit as this is the minimum number denoted in the spec.